[5093] Remove state machine from TextfieldPropertySection

mcharfadi opened the pull request.

frouene commented:
I've found a regression with the new version, when editing a number textfield with a non-number entry, the value is not removed after lost focus.
(see on the left, comparing the existant on the right)
![textfield-regression](https://github.com/user-attachments/assets/9aa88a06-bf3e-4901-990f-ead1a7e3f780)

frouene reviewed (changes requested):
Regression with number textfield

frouene commented:
It seems ok now.
Can you add cypress tests, in particular to ensure non-regression on autocompletion, submit on focus lost, and on the number textfield validation.

frouene commented:
On completion without result, we still display an empty box 
<img width="452" height="492" alt="image" src="https://github.com/user-attachments/assets/4bfca4ba-7631-4e85-946d-3b98367f4673" />

mcharfadi committed:
[5093] Remove state machine from TextfieldPropertySection

Bug: https://github.com/eclipse-sirius/sirius-web/issues/5093
Signed-off-by: Michaël Charfadi <michael.charfadi@obeosoft.com>

frouene commented on integration-tests/cypress/e2e/project/details/textfield-auto-completion.cy.ts:
```diff
@@ -0,0 +1,57 @@
+          details.getTextField('Domain Type').contains(actualValue as string);
+        });
+
+      // const firstElementText : string = cy.getByTestId('completion-proposals').children[0];
```
dead code

frouene commented:
A new strange behaviour with the auto completion feature, juste after use it to add some text if you press `space` (without `ctrl`) proposals appear (it's the previous behaviour, don't know if that's what's intended, though.)
But with the new version when try to add a proposal nothing happen.

![textfield-proposal](https://github.com/user-attachments/assets/b97ad978-d34d-40d7-bdb9-5892ace1e991)

mcharfadi commented:
> A new strange behaviour with the auto completion feature, juste after use it to add some text if you press `space` (without `ctrl`) proposals appear (it's the previous behaviour, don't know if that's what's intended, though.) But with the new version when try to add a proposal nothing happen.
> 
I don't think that this was supposed to happened. I think it was an issue with how we handled `ctrl+space` with a state to know if ctrl was pressed and it seems that the state was still true. 
I changed the listener to use` event.ctrlKey`

Also note that pressing the left or right arrow was not working on master so I removed this part.
We can still select a value with the keyboard by pressing tab and then use the arrows.

mcharfadi commented:
I added the `autofocus `prop to the `ProposalsList ` `MenuList `so we don't have to press tab before using the arrows.

frouene commented:
> I added the `autofocus `prop to the `ProposalsList ` `MenuList `so we don't have to press tab before using the arrows.

Yes, auto-completion is more pleasant to use with these changes

frouene commented:
Could be merged once the test updated
